### ISSUE TITLE
FR: MongoDB support `Infinity` data type

### ISSUE THREAD

**Issue by aladinflux** [prisma/prisma#12891]
## Problem

MongoDB supports storing `Infinity` as a value. We store that in MongoDB to refer to "unlimited" value.

## Suggested solution

Allow Prisma to support the `Inifinity` data type.

## Alternatives

We're now storing a ridiculously large number (as a project wide constant) as an alternative. We could've used `null` but it will be difficult to do queries on it.

## Additional context

For example: find all users with allowed number of posts below 1 million. In this case, we can store the users with unlimited posts with `null`, but it means we need to always query: where x > 100 or where x === null. For now, we just store a constant of a very large number for such cases.

**Comment by janpio** [comment]
Is this really a data type, or just a value for other types? 
Can you share a data example of MongoDB of this value in action?

**Comment by aladinflux** [comment]
You are right @janpio , it's not a data type, but rather a value of `Double` apparently. It seems like `Double` not only supports `Infinity` and `-Infinity`, but also `NAN`.

Here's a link, just search for `Infinity`: https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/

As I mentioned, a use case would be for example having a list of users who have a limit on the number of comments or posts they can make. With MongoDB, one can filter users based on that value:

`db.collection('users').find({ postsLimit: { $gt: 100 } })` and MongoDB will include users who have `Infinity` as a value. However, it won't include `-Infinity`.
We can also query:
`db.collection('users'),find({ postsLimit: Infinity })` to filter only those who can post unlimited data.

Without Infinity, we will need to store the value as `null` (which doesn't make sense) or add another value `canPostUnlimited: true` and this will result in additional clauses in the query:
`db.collection('users').find({ $or: [{ postsLimit: { $gt: 100 }, { canPostUnlimited: true }] )`

As a last resort, one can specify an arbitrary large number as an app-wide constant `const LARGE_NUMBER = 999999999999999` and use that in queries instead of `Infinity`, but I feel that this approach is not safe.

For `NAN`, I did not personally have any use for it, but maybe some will as it's a valid supported value on MongoDB.
